### ISSUE TITLE
Dynamic access to case class properties

### ISSUE THREAD

**Issue by behrad** [milessabin/shapeless#281]
Taking I have

```scala
case class Book(author: String, title: String, id: Int, price: Double)
```

I want to be able to get a field's value in runtime having field's name as String:
```scala
val b = Book(....)
b.get( "author" )
```

How could this be achieved in Shapeless?

I wrote

```scala
object test {
        def f() = {
          import shapeless._
          import record._
          import ops.record._
          import syntax.singleton._

          case class Book(author: String, title: String, id: Int, price: Double)
          val bookGen = LabelledGeneric[Book]
          val tapl = Book("Benjamin Pierce", "Types and Programming Languages", 262162091, 44.11)
          val rec = bookGen.to(tapl)
          println( "=================================== " + rec('price) )
        }
      }
      test.f()
```

but got:
```
 scala.ScalaReflectionException: <none> is not a term
[error] 	at scala.reflect.api.Symbols$SymbolApi$class.asTerm(Symbols.scala:259)
[error] 	at scala.reflect.internal.Symbols$SymbolContextApiImpl.asTerm(Symbols.scala:73)
[error] 	at shapeless.GenericMacros$Helper$class.mkTransCase(generic.scala:393)
[error] 	at shapeless.GenericMacros$$anon$1.mkTransCase(generic.scala:87)
[error] 	at shapeless.GenericMacros$Helper$class.mkToProductReprCase(generic.scala:404)
[error] 	at shapeless.GenericMacros$$anon$1.mkToProductReprCase(generic.scala:87)
[error] 	at shapeless.GenericMacros$Helper$$anonfun$5.apply(generic.scala:431)
[error] 	at shapeless.GenericMacros$Helper$$anonfun$5.apply(generic.scala:431)
[error] 	at shapeless.GenericMacros$Helper$class.mkProductCases(generic.scala:334)
[error] 	at shapeless.GenericMacros$$anon$1.mkProductCases(generic.scala:87)
[error] 	at shapeless.GenericMacros$Helper$class.materializeGeneric(generic.scala:431)
[error] 	at shapeless.GenericMacros$$anon$1.materializeGeneric(generic.scala:87)
[error] 	at shapeless.GenericMacros$.materializeAux(generic.scala:99)
[error] 	at shapeless.GenericMacros$.materializeLabelled(generic.scala:74)
[error]           val bookGen = LabelledGeneric[Book]

```

**Comment by milessabin** [comment]
Could you give this another try with the latest 2.2.0-SNAPSHOT or 2.2.0-RC1?

**Comment by milessabin** [comment]
This appears to be fixed in 2.3.1.
